api/controllers/v0.3/endpoints.js

```diff
@@ -44,6 +44,30 @@ const envVariableNames = [
   "SO_LICENSE_KEY",
 ];
 
+/**
+ * checkIlsToken
+ * check to see if the ILS token is available or if it is expired. If either is
+ * true, then request a new one. If there was an issue, return an error. Returns
+ * an array [boolean, message] with information if the ILS token generation
+ * call was successful or not.
+ */
+async function checkIlsToken(req, res) {
+  // If the ilsClient has no token or the token is expired, then
+  // generate a new token.
+  if (!ilsClient.hasIlsToken() || ilsClient.isTokenExpired()) {
+    try {
+      await ilsClient.generateIlsToken();
+      return [false, null];
```

Comment:
Maybe it's worth returning a more informative message there?  (Or will no one ever see it?)

Reply:
Right, no one will ever see it. If the first value is true, then the second value is returned.